Sources/Rendering/Core/ColorTransferFunction/CssFilters.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * A helper file to transform RGBA points using CSS filters equivalent
+ * The equivalents of CSS filters using SVG filters can be found here:
+ * https://www.w3.org/TR/filter-effects-1/#ShorthandEquivalents
+ * For each SVG filter, you can look for the maths behind it on the same page:
+ * https://www.w3.org/TR/filter-effects-1/#FilterPrimitivesOverview
+ *
+ * For example, the saturate filter equivalent is here:
+ * https://www.w3.org/TR/filter-effects-1/#saturateEquivalent
+ * And the maths behind the feColorMatrix of type saturate is here:
+ * https://www.w3.org/TR/filter-effects-1/#ref-for-attr-valuedef-type-saturate
+ *
+ * The transforms are done using matrices of size 5 by 5. They are row major
+ * as in vtkMath. The vectors representing the RGBA points uses
+ * [R, G, B, A, 1] vectors, with each channel between 0 and 1.
+ */
+
+import { identity, multiplyMatrix } from 'vtk.js/Sources/Common/Core/Math';
+
+export const luminanceWeights = [0.213, 0.715, 0.072];
+
+export function getNewFilter() {
+  return new Array(25);
+}
+
+export function getIdentityFilter(outFilter = getNewFilter()) {
```

Comment:
isn't `getNewFilter()` always called even when an `outFilter` parameter is passed?
If so, then `return identity(5, outFilter || getNewFilter());` might be more appropriate

Reply:
No, it is called only if no parameter is given
Here is a code snippet to test it:
```js
function foo() {
  console.log("foo");
  return 42;
}

function bar(x = foo()) {
  return x + 10;
}

bar(); // prints: foo
bar(2); // nothing is printed
```